src/sst/core/configGraph.cc

```diff
@@ -381,6 +404,64 @@ ConfigComponent* ConfigComponent::findSubComponentByName(const std::string& name
     return nullptr;
 }
 
+ConfigStatistic* ConfigComponent::addStatistic(StatisticId_t sid, const std::string& statisticName)
+{
+    // Check for Enable All Statistics
+    if (statisticName == STATALLFLAG) {
+        // Force the STATALLFLAG to always be on the bottom of the list.
+        // First check to see if anything is in the vector, if vector is empty,
+        // a STATALLFLAG flag will be added to the vector
+        if (false == enabledStatistics.empty()) {
+            // The vector is populated, so see if the STATALLFLAG
+            // already exists if it does, we are done
+            if (STATALLFLAG != enabledStatistics.back().name) {
+                // Add a STATALLFLAG to end of the vector
+                enabledStatistics.emplace_back(STATALLFLAG);
+                return &(enabledStatistics.back());
+            }
+        } else {
+            // Add a STATALLFLAG to end of the vector
+            enabledStatistics.emplace_back(STATALLFLAG);
+            return &(enabledStatistics.back());
+        }
+    } else {
+
+      /* Check for existing statistic with this name */
+      for ( auto &i : enabledStatistics ) {
+          if ( i.name == statisticName)
+          {
+              return nullptr;
+          }
+      }
+
+      enabledStatistics.emplace(enabledStatistics.begin(),
+          ConfigStatistic(sid, id, statisticName));
+
+
+      return &(enabledStatistics.front());
+    }
+
+    return nullptr;
+}
+
+ConfigStatistic* ConfigComponent::findStatistic(StatisticId_t sid)
+{
+    return const_cast<ConfigStatistic*>(const_cast<const ConfigComponent*>(this)->findStatistic(sid));
```

Comment:
I think we can remove the const casts here. See comment below.

Reply:
I don't think that makes sense to have both : 
ConfigStatistic* ConfigComponent::findStatistic(StatisticId_t sid)
ConfigStatistic* ConfigComponent::findStatistic(StatisticId_t sid) const

I can remove the const_casts but I think that now we have a non const returned object on the const method, we can simply remove the first non const method.

👍 99d5b75b93fdc02bbd121925d0582eacb1e87e2e